airflow/models/taskinstance.py

```diff
@@ -883,6 +885,12 @@ def _run_raw_task(
                 context = self.get_template_context()
 
                 task_copy = copy.copy(task)
+
+                if issubclass(type(task_copy), BaseSensorOperator) and \
+                        conf.get('core', 'executor') in ("InProcessExecutor", "SequentialExecutor"):
+                    self.log.warning("InProcessExecutor changes sensor mode to 'reschedule'.")
+                    task_copy.mode = 'reschedule'
+
```

Comment:
This change allow us to run DAGs with sensors using `InProcessExecutor` and `SequentialExecutor`. To be honest I think we can consider replacing sequential with inprocess. 

@potiuk @mik-laj @evgenyshulman any thoughts about that?